submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,140 @@
+class Creature {
+    constructor({ species, name, gender, saying, friends }) {
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+    }
+    returnFullInfo() {
+        const keys = [
+            "species",
+            "name",
+            "gender",
+            "legs",
+            "hands",
+            "paws",
+            "tail",
+            "saying",
+            "friends",
+        ];
+        return keys
+            .reduce((output, key) => {
+                if (this[key]) {
+                    if (typeof this[key] === "function")
+                        return output.concat(this[key]());
+                    else return output.concat(this[key]);
+                } else return output;
+            }, [])
+            .join(", ");
+    }
+}
+class Animal extends Creature {
+    constructor(props) {
```

Comment:
As a peer developer I want to instantly know what properties can be without digging into the code too much. Besides Intellisense won't be really helpful when method's signature is so secretive.